### PR TITLE
adds new set of custom metrics for ingestion pipeline

### PR DESCRIPTION
**Pull request by kotharironak**
As part of an issue (https://github.com/hypertrace/hypertrace-ingester/issues/109) on filling gaps on monitoring hypertrace, it adds the following metrics in the ingestion pipeline. Also made a modification on the naming convention of existing metrics to start with all metrics with `hypertrace`
- hypertrace_reported_spans_total: Counter for measuring received spans, dropped and error
- hypertrace_reported_spans_processed_total: Counter to measure the number of spans processed per tenant group
- hypertrace_emitted_traces_total: Counter to measure the number of traces emitted
- hypertrace_enriched_traces_total: Counter to measure the number of traces enriched
- hypertrace_trace_enrichment_latency: Timer to measure the time taken in the enrichment process

### CODE REVIEW

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java [hunk 1]
```diff
@@ -152,7 +152,7 @@ public RawSpan convert(Span jaegerSpan) throws Exception {
             tenantId.get(),
             tenant ->
                 PlatformMetricsRegistry.registerTimer(
-                    SPAN_NORMALIZATION_TIME_METRIC, Map.of("tenantid", tenant)))
+                    SPAN_NORMALIZATION_TIME_METRIC, Map.of("tenantId", tenant)))
```
Comment (kotharironak) [inline span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java:155]: note: @samarth-gupta-traceable we may have to make changes in our dashboard.

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java [hunk 2]
```diff
@@ -24,9 +24,13 @@
   private static final Logger LOGGER = LoggerFactory
       .getLogger(JaegerSpanToAvroRawSpanTransformer.class);
 
-  private static final String VALID_SPAN_RECEIVED_COUNT = "valid.span.received.count";
```
Comment (kotharironak) [inline span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java:27]: note: @samarth-gupta-traceable we have to make changes in our dashboard

**Review by kotharironak — COMMENTED** [review]
@surajpuvvada why do we need `mutex` in this method - recordSpansPerTrace of TraceEmitPunctuator?

**Comment by kotharironak** [comment]
Tested locally,
Screenshot 2021-01-21 at 1 23 55 PM
Screenshot 2021-01-21 at 1 24 15 PM
Screenshot 2021-01-21 at 1 23 33 PM
Screenshot 2021-01-21 at 1 23 11 PM
Screenshot 2021-01-21 at 1 23 00 PM
Screenshot 2021-01-21 at 1 27 22 PM

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java [hunk 3]
```diff
@@ -24,9 +24,13 @@
...
+  private static final String SPANS_COUNTER = "hypertrace.reported.spans";
+  private final ConcurrentMap<String, Counter> statusToSpansCounter = new ConcurrentHashMap<>();
```
Comment (rish691) [inline span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java:28]: Do we need concurrent map? The processor code is executed per stream thread.

Comment (kotharironak) [reply span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java:28]: Make sense, will convert it to a static map as they are referring to the same metric.

Comment (kotharironak) [reply span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java]: done

**Review by rish691 — APPROVED** [review]
LGTM

**Review by JBAhire — APPROVED** [review]

File: span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java (hunk 1 above)
Comment (samarthgupta437) [reply span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java:155]: got it .

**Review by laxmanchekka — APPROVED** [review]

**Pull request merged**